include/aws/http/request_response.h

```diff
@@ -194,6 +194,42 @@ typedef void(aws_http_on_stream_complete_fn)(struct aws_http_stream *stream, int
  */
 typedef void(aws_http_on_stream_destroy_fn)(void *user_data);
 
+/**
+ * Tracing metrics for aws_http_stream.
+ * Data maybe not be available if the data of stream was never sent/received before it completes.
+ */
+struct aws_http_stream_metrics {
+    /* The time stamp when the request started to be sent to HTTP level. 0 means data not available. */
+    uint64_t send_start_timestamp_ns;
```

Comment:
nit: its a bit unclear what http level refers to here? time when we gave the first byte of request to os networking stack? or something else?

Reply:
I was planned to use the time stamp sent to network stack, but it's awkward to do for h2. Since we may invoke the complete callback for h2 stream before any of the frame sent to the network stack, when the stream reset by client before any network stuff happens.

So, I changed to be `sent from http stream`, which is the time when stream encode the frame/data to the connection to send. For h1, it's the same as sending to the networking stack.

Reply:
I am bit confused about the difference between `from http stream` vs `http level`. It might be helpful to add some documentation about what it means by `from http stream`. In the case, where we may invoke complete callback before any of the frame sent to the network stack, I would expect the data to be not available.

Receive timestamp is still from http level, maybe forgot to update?